server/src/main/java/com/microsoft/java/bs/core/internal/managers/BuildTargetManager.java

```diff
@@ -109,17 +122,49 @@ private List<String> getBuildTargetTags(boolean hasTests) {
     return tags;
   }
 
-  private void setJvmBuildTarget(GradleSourceSet sourceSet, BuildTarget bt) {
+  private void setBuildTarget(GradleSourceSet sourceSet, BuildTarget bt) {
+    if (sourceSet.isKotlin()) {
```

Comment:
If a source set applies kotlin, scala & java plugins, then which one should be returned as the data of the build target?

I've not idea about this. I asked this question in the BSP discord channel.

Reply:
I've no idea either.  I doubt it's possible.  I think that both the Scala and Kotlin plugins are Java aware and able to mix in their languages with Java so you could have Java classes referencing Scala classes and vice versa in the same project.  But neither plugin is aware of the other language so Scala wouldn't know what to do with Kotlin and Kotlin wouldn't know what to do with Scala.  Happy to be corrected on this but I think the code would have to be in separate projects.

Reply:
Searched in google and found that there are some people doing that: https://discuss.gradle.org/t/kotlin-groovy-and-java-compilation/14903/8.

But I guess that's a very rare case.